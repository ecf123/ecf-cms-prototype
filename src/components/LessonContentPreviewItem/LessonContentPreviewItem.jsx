import "./LessonContentPreviewItem.scss"
import binIcon from "../../assets/bin-icon.svg";  
import penIcon from "../../assets/pen-icon.svg";

const LessonContentPreviewItem = ({lesson,handleEdit,handleDelete}) => {

  const {title, content} = lesson ?? {};

  return (
    <div className="lesson-content__item">
      <h6 className="lesson-content__title">{title}</h6>
      <p className="lesson-content__text">{content}</p>
      <div className="buttons">
        <img
          className="buttons__bin"
          src={binIcon}
          alt="Delete Question"
          onClick={handleDelete}
        />
        <img
          className="buttons__pen"
          src={penIcon}
          alt="Edit Question"
          onClick={handleEdit}
        />
      </div>
    </div>
  )
}

export default LessonContentPreviewItem