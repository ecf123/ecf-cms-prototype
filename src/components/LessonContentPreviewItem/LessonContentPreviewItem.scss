@use "../../styles/_typography.scss" as *;
@use "../../styles/_variables.scss" as *;

.lesson-content {
  &__item {
    width: fit-content;
    height: fit-content;
    font-size: $font-size-extra-small;
  }

  &__title {
    font-weight: $font-weight-bold;
    color: $color-blue-navy-dark;
  }

  &__text {
    font-weight: $font-weight-regular;
    color: $color-grey-normal;

    margin-top: 0.56rem;
    margin-bottom: 1.4rem;
  }
}

.buttons {
  display: flex;
  gap: 1rem;
  cursor: pointer;
  &__pen {
    width: 1rem;
    height: 1rem;
    filter: brightness(0) saturate(100%) invert(44%) sepia(16%) saturate(518%)
      hue-rotate(182deg) brightness(94%) contrast(86%);
  }
  &__bin {
    width: 1rem;
    height: 1rem;
    filter: brightness(0) saturate(100%) invert(31%) sepia(78%) saturate(1774%)
      hue-rotate(324deg) brightness(85%) contrast(88%);
  }
}
