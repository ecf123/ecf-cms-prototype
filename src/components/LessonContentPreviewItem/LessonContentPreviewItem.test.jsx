import LessonContentPreviewItem from "./LessonContentPreviewItem";
import { customRender } from "../../utils/testUtils";

const lesson = {
  title: "Test title",
  content: "Content"
}

it("Should render the preview item with props", () => {
  const { container } = customRender(<LessonContentPreviewItem lesson={lesson} />);
  expect(container).toMatchSnapshot();
});

it("Should render the preview item without props", () => {
  const { container } = customRender(<LessonContentPreviewItem />);
  expect(container).toMatchSnapshot();
});