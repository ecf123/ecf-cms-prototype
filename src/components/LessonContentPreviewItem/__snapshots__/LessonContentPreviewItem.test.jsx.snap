// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the preview item with props 1`] = `
<div>
  <div
    class="lesson-content__item"
  >
    <h6
      class="lesson-content__title"
    >
      Test title
    </h6>
    <p
      class="lesson-content__text"
    >
      Content
    </p>
    <div
      class="buttons"
    >
      <img
        alt="Delete Question"
        class="buttons__bin"
        src="bin-icon.svg"
      />
    </div>
  </div>
</div>
`;

exports[`Should render the preview item without props 1`] = `
<div>
  <div
    class="lesson-content__item"
  >
    <h6
      class="lesson-content__title"
    />
    <p
      class="lesson-content__text"
    />
    <div
      class="buttons"
    >
      <img
        alt="Delete Question"
        class="buttons__bin"
        src="bin-icon.svg"
      />
    </div>
  </div>
</div>
`;
